### PR TITLE
Display parent availability for FOLIO bound-with children

### PR DESCRIPTION
**Pull request by marlo-longley**
Closes https://github.com/sul-dlss/SearchWorks/issues/3494

### CODE REVIEW

File: app/components/access_panels/location_item_component.html.erb [hunk 1]
```diff
@@ -10,7 +10,7 @@
     </td>
   <% end %>
   <% if render_item_details? %>
-    <td class="item-availability" data-live-lookup-id="<%= document.live_lookup_id %>" data-status-target=".availability-icon" data-item-id="<%= item.live_lookup_item_id if item.live_status? %>" <%= "data-request-url='#{helpers.request_url(document, library: item.library, location: item.home_location, barcode: item.barcode)}'".html_safe if render_real_time_availability_request_link? %>>
+    <td class="item-availability" data-live-lookup-id="<%= item.live_lookup_instance_id %>" data-status-target=".availability-icon" data-item-id="<%= item.live_lookup_item_id if item.live_status? %>" <%= "data-request-url='#{helpers.request_url(document, library: item.library, location: item.home_location, barcode: item.barcode)}'".html_safe if render_real_time_availability_request_link? %>>
```
Comment (jcoyne) [inline app/components/access_panels/location_item_component.html.erb:13]: Do we also need to change this line? https://github.com/sul-dlss/SearchWorks/blob/master/app/components/access_panels/at_the_library_component.html.erb#L2 ?

Comment (cbeer) [reply app/components/access_panels/location_item_component.html.erb:13]: I don't think so (although maybe it's redundant? Or we should be omitting item-level attributes if it's the same as that?)

File: lib/holdings/item.rb [hunk 2]
```diff
@@ -134,8 +133,17 @@ def stackmapable?
       stackmapable_library? && stackmapable_location?
     end
 
+    def bound_with_parent
+      return nil unless document && document['holdings_json_struct']
+
+      match = document['holdings_json_struct'].first['holdings']&.find do |item|
+        item.dig('boundWith', 'item', 'barcode') == barcode
```
Comment (cbeer) [inline lib/holdings/item.rb:140]:
```suggestion
        item.dig('boundWith', 'item', 'id') == id
```

I think we've seen items without barcodes or duplicated barcodes (😬). `id` ought to always be available though.

File: lib/holdings/item.rb [hunk 3]
```diff
@@ -134,8 +133,17 @@ def stackmapable?
       stackmapable_library? && stackmapable_location?
     end
 
+    def bound_with_parent
+      return nil unless document && document['holdings_json_struct']
```
Comment (cbeer) [inline lib/holdings/item.rb:137]:
```suggestion
      return nil unless document && document['holdings_json_struct'].any?
```

I think an empty array would also evaluate to `true` and break with a NoMethod error below, right?

File: lib/holdings/item.rb [hunk 4]
```diff
@@ -134,8 +133,17 @@ def stackmapable?
...
+
+      match = document['holdings_json_struct'].first['holdings']&.find do |item|
```
[the remark this answers is not in the document]
Comment (marlo-longley) [reply lib/holdings/item.rb]: @cbeer not sure I did what you meant, but I moved some login into `folio_holdings`

**Pull request merged**